pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1438,3 +1446,269 @@ func (vs *VSphere) GetZoneByNodeName(ctx context.Context, nodeName k8stypes.Node
 func (vs *VSphere) GetZoneByProviderID(ctx context.Context, providerID string) (cloudprovider.Zone, error) {
 	return cloudprovider.Zone{}, cloudprovider.NotImplemented
 }
+
+// GetLabelsForVolume implements the PVLabeler interface for VSphere
+// since this interface is used by the PV label admission controller.
+func (vs *VSphere) GetLabelsForVolume(ctx context.Context, pv *v1.PersistentVolume) (map[string]string, error) {
+	// ignore if zones not enabled
+	if !vs.isZoneEnabled() {
+		klog.V(4).Infof("Zone labels for volume is not enabled in vsphere.conf")
+		return nil, nil
+	}
+	// ignore if not vSphere volume
+	if pv.Spec.VsphereVolume == nil {
+		return nil, nil
+	}
+	return vs.GetVolumeLabels(pv.Spec.VsphereVolume.VolumePath)
+}
+
+// GetVolumeLabels returns the well known zone and region labels for given volume
+func (vs *VSphere) GetVolumeLabels(volumePath string) (map[string]string, error) {
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Move this code to after zones check as this is redundant if zones is not enabled.